### PR TITLE
LPD-37767 Greater than filter for the dateModified fiels does not work properly with custom objects in MySQL

### PR DESCRIPTION
**Pull request by dannielraposo**
**What's changed?**
This PR addresses the issue of date format mismatch when querying MySQL. Specifically, when filtering fields of type `DATE_TIME` (defined in the entity type), the query now ensures the retrieved values are formatted correctly, removing microseconds to align with the OData format used for filtering. This change ensures consistency in date comparisons and resolves filtering discrepancies.

See the following Jira Ticket: [LPD-37767)](https://liferay.atlassian.net/browse/LPD-37767)

TODO:

- [ ] Adapt tranformer for all DBs.
- [ ] Analyze business type (`DATE`) and entity type (`DATE_TIME`) discrepancy for `modifiedDate` and `createDate` fields)
- [ ] Fix similar filtering issue for fields with business type set to `DATE_TIME`

### CODE REVIEW

**Comment by dannielraposo** [comment]
ci:test:relevant

**Review by 4lejandrito — CHANGES_REQUESTED** [review]
You can squash the last 2 commits into a single one.

**Comment by 4lejandrito** [comment]
Also @dannielraposo we need to add an integration test that reproduces the original issue and is fixed by your changes. The unit test is not enough because it does not expose the issue.

**Comment by dannielraposo** [comment]
Failing tests are not related to the PR

File: modules/core/petra/petra-sql-dsl-spi/src/main/java/com/liferay/petra/sql/dsl/spi/expression/DSLFunctionType.java [hunk 1]
```diff
@@ -29,6 +29,9 @@ public class DSLFunctionType {
 	public static final DSLFunctionType CONCAT = new DSLFunctionType(
 		"CONCAT(", ")");
 
+	public static final DSLFunctionType DATE2SEC = new DSLFunctionType(
```
Comment (4lejandrito) [inline modules/core/petra/petra-sql-dsl-spi/src/main/java/com/liferay/petra/sql/dsl/spi/expression/DSLFunctionType.java:32]: What about `TRUNCATE_TO_SECONDS`? 

cc @ccorreagg

**Review by 4lejandrito — CHANGES_REQUESTED** [review]

**Review by 4lejandrito — APPROVED** [review]

**Comment by dannielraposo** [comment]
ci:test:relevant

**Comment by dannielraposo** [comment]
ci:forward

**Comment by dannielraposo** [comment]
ci:forward:force

File: portal-impl/test/unit/com/liferay/portal/dao/sql/transformer/BaseSQLTransformerLogicTestCase.java [hunk 2]
```diff
@@ -240,7 +240,7 @@ protected String getCrossJoinTransformedSQL() {
 	}
 
 	protected String getDateFormatOriginalSQL() {
-		return "select foo from Foo where DATE2SEC(foo)";
+		return "select foo from Foo where TRUNCATE_TO_SECONDS(foo)";
```
Comment (ccorreagg) [inline portal-impl/test/unit/com/liferay/portal/dao/sql/transformer/BaseSQLTransformerLogicTestCase.java:243]: I think we should add a date of example, this command does not exist even if we ignore the TRUNCATE_TO_SECONDS :thinking: 

Something like:

`select foo from Foo where TRUNCATE_TO_SECONDS(foo) = '2024-10-04T09:03:00Z'`

**Comment by dannielraposo** [comment]
ci:forward:force

**Comment by dannielraposo** [comment]
ci:forward:force

**Comment by ccorreagg** [comment]
ci:stop

**Comment by dannielraposo** [comment]
ci:forward

**Comment by dannielraposo** [comment]
ci:stop

**Comment by dannielraposo** [comment]
ci:test:sf

**Comment by dannielraposo** [comment]
ci:forward:force

**Comment by dannielraposo** [comment]
ci:forward:force

**Pull request closed without merging**